BlueShift-iOS-Extension-SDK/BlueShiftPushNotification.m

```diff
@@ -210,62 +210,88 @@ - (NSArray *)mediaAttachmentDownlaod:(UNNotificationRequest *)request {
+                    if([self isCatgoryExists:category.identifier inSet:updatedCategories] == NO) {
+                        [updatedCategories addObject:category];
+                        [[UNUserNotificationCenter currentNotificationCenter] setNotificationCategories:updatedCategories];
+                    } else if(forceReplaceCategory && [forceReplaceCategory boolValue] == YES) {
+                        // Remove old category with same id(if present) in order to replace it.
+                        [self removeDuplicateCategory:category.identifier fromSet:updatedCategories];
+                        [updatedCategories addObject:category];
+                        [[UNUserNotificationCenter currentNotificationCenter] setNotificationCategories:updatedCategories];
+                    }
+                    // set the flag to true to exit the thread sleep loop.
+                    isCategoryRegistrationComplteted = YES;
+                }];
+                int counter = 0;
+                // Sleep thread till the category registration finishes or counter reaches to 20 (2 seconds)
+                while (isCategoryRegistrationComplteted == NO && counter < kThreadSleepIterations) {
+                    counter++;
+                    [NSThread sleepForTimeInterval:kThreadSleepTimeInterval];
+                }
             }
         } @catch (NSException *exception) {
         }
     }
 }
 
-- (NSMutableArray*)getNotificationActions:(NSArray*) actionsDict {
+- (NSMutableArray*)getNotificationActions:(NSArray*)actions {
     NSMutableArray<UNNotificationAction *>* notificationActions = [NSMutableArray new];
-
-    for(NSDictionary* actionItem in actionsDict) {
-        UNNotificationActionOptions actionOption = UNNotificationActionOptionForeground;
-        NSString* actionType = actionItem[kNotificationActionType];
-        if (actionType && ![actionType isEqualToString:kNotificationActionTypeOpen]) {
-            if([actionType isEqualToString:kNotificationActionTypeDestructive])
-                actionOption = UNNotificationActionOptionDestructive;
-            else if([actionType isEqualToString: kNotificationActionTypeDestructive])
-                actionOption = UNNotificationActionOptionAuthenticationRequired;
-            else if([actionType isEqualToString:kNotificationActionTypeNone])
-                actionOption = UNNotificationActionOptionNone;
-        }
-        if (actionItem[kNotificationActionTitle] && actionItem[kNotificationActionIdentifier]) {
-            UNNotificationAction* action = [UNNotificationAction actionWithIdentifier:actionItem[kNotificationActionIdentifier] title:actionItem[kNotificationActionTitle] options:actionOption];
-            [notificationActions addObject:action];
+    if (actions) {
+        @try {
+            // Check to support maximum 5 action items
+            for(int counter = 0; (counter < actions.count && counter < kNotificationMaxSupportedActions); counter++) {
```

Comment:
instead of doing this 2 check each time, can we take min of these two values and use it here?